### PR TITLE
[RSDK-3634] Implement data.proto APIs that Add/Remove Tags

### PR DESCRIPTION
**Pull request by 8ashar**
This PR implements and adds docstrings to the gRPC methods that'll allow a DataClient to send a gGRPC request to app to add, remove, and retrieve tags. The PR also implements a function to retrieve bounding box labels. There are a few other minor changes throughout. Namely, some function signatures are updated to include defaults for certain parameters (i.e. empty filter), and the AppClient "close()" function is implemented with logging.

### CODE REVIEW

File: src/viam/app/client.py
Comment (njooma) [inline src/viam/app/client.py:52]: We probably don't need this, since the line above says the same thing

File: src/viam/app/data/client.py
Comment (njooma) [inline src/viam/app/data/client.py:79]: Should these be optional? Seems like adding tags to binary data by file ids should REQUIRE the file_ids and the tags

File: src/viam/app/data/client.py
Comment (njooma) [inline src/viam/app/data/client.py:90]: Same, should these be required?

Also, `Args` order should be in the same order as in the method signature

File: src/viam/app/data/client.py
Comment (njooma) [inline src/viam/app/data/client.py:101]: Looks like we're going to run into the `Optional` vs `Required` question on all of these.

If these truly are optional, the docstring should mention what the behavior is when left empty.

**Review by njooma — CHANGES_REQUESTED** [review]

File: src/viam/app/data/client.py
Comment (8ashar) [reply src/viam/app/data/client.py:79]: See belo

File: src/viam/app/data/client.py
Comment (8ashar) [reply src/viam/app/data/client.py:101]: See above

**Review by stuqdog — CHANGES_REQUESTED** [review]
One nit, but on first pass looks good!

**Review by njooma — APPROVED** [review]

**Comment by stuqdog** [comment]
@8ashar per this PR, `{add|remove}_tags_by_file_id` is going to become `..._by_id` instead. Might be good to change the user-facing method names now, just so it's always consistent.

**Review by stuqdog — CHANGES_REQUESTED** [review]
LGTM but I'm requesting that we change the user-facing API of the `*_by_file_ids` methods to instead be `*_by_ids`, and update the args/comments appropriately. The methods are changing name and we should fix the user-facing experience to match right away, then update the methods once the new protos go into effect.

See [this](https://github.com/viamrobotics/api/pull/281) pr for details.

Edit: in addition, we should change the actual argument structure. Right now we're passing file IDs (which are strings) into a deprecated `file_id` field that isn't looked at. What we should pass are BinaryIDs (which are a proto type `BinaryID` and which include, among other things, an optional `file_id`) into the `binary_id` field that _is_ looked at.

File: src/viam/app/client.py
Comment (stuqdog) [inline src/viam/app/client.py:27]: Looks like this reverted back to `viam.app.com`, we should change it back.

File: src/viam/app/data/client.py
Comment (stuqdog) [inline src/viam/app/data/client.py:23]: We probably want to still have this line.

File: src/viam/app/data/client.py
Comment (stuqdog) [inline src/viam/app/data/client.py:50]: I think we don't need this, since `close` is now being handled by the `AppClient`.

File: src/viam/app/client.py
Comment (8ashar) [reply src/viam/app/client.py:21]: Thanks for picking this out I think I accidentally reset a commit or two too far back :/.

File: src/viam/app/data/client.py
Comment (stuqdog) [inline src/viam/app/data/client.py:173]: Awesome to see that you went ahead and implemented these two! We should add docstring comments explaining the semantics of these a bit. Also, have you tested them at all? Probably good to do so.

It's also totally fine to leave these for a subsequent PR, and in fact maybe good to do so if you think that testing/documenting might take some time.

**Review by stuqdog — COMMENTED** [review]
One comment that can definitely be pushed to a subsequent PR. Also: we should hold off on merging this until the data proto updates have been merged.

File: src/viam/app/data/client.py
Comment (8ashar) [reply src/viam/app/data/client.py:173]: As per my Slack message: I meant to leave this out of this PR! The functions work and upload data but trying to decipher the required parameters is its own mess that is probably better solved in the data_sync.proto ticket. I think I'm going to revert these back to raising unimplemented errors and commit again. I have a good local testing suite set up for these though that highlights what we might want to include

**Review by stuqdog — APPROVED** [review]
We should coordinate release timing between this, the other data API PR, and the proto updates, but LGTM! Thanks for pulling this together.

File: src/viam/app/data/client.py
Comment (stuqdog) [inline src/viam/app/data/client.py:89]: oh tiny nit here, `s/AddTagstoBinaryDataByIDsResponse/AddTags**T**oBinaryDataByIDsResponse`

Comment (8ashar) [reply src/viam/app/data/client.py:89]: oouh nice catch

**Pull request merged**
